docs for the HTTP transport adapter (#1285)
Azure/azure-sdk-for-cpp: CONTRIBUTING.md

# Use -R to use a regular exp for what to run
ctest -R Http # runs only Http tests
```

> The Azure SDK for C++ supports libcurl and WinHTTP libraries as HTTP stacks and provides an HTTP transport adapter for each one. It can also support an external, `customer-implemented` *HTTP transport adapter*. To learn more about this feature, [click here](LinkForTransportAdapterDocs).

Reviewer: https://dev.azure.com/azure-sdk/public/_build/results?buildId=678382&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=ccad100f-2904-5913-2083-c525ad929a03&l=52
Looks like link verification is causing errors:
> Summary of broken links:
'file:///D:/a/1/s/sdk/core/azure-core/README.md' has 1 broken link(s):
  LinkForTransportAdapterDocs
'file:///D:/a/1/s/CONTRIBUTING.md' has 1 broken link(s):
  LinkForTransportAdapterDocs
'file:///D:/a/1/s/doc/HttpTransportAdapter.md' has 1 broken link(s):
  #Building-a-custom-HTTP-Transport-Adapter

Also two warnings:
> ##[warning]Please lower case your anchor tags (i.e. anything after '#' in your link '#Building-a-custom-HTTP-Transport-Adapter'. Check here for more information: https://aka.ms/azsdk/guideline/links

> ##[warning]DO NOT use relative link LinkForTransportAdapterDocs. Please use absolute link instead. Check here for more information: https://aka.ms/azsdk/guideline/links

> Out of the box, the Azure SDK for C++ supports the libcurl and WinHTTP libraries as HTTP stacks for communicating with Azure services over the network and provide an HTTP transport adapter for each one. It can also support an external, `customer-implemented` *HTTP transport adapter* based on other HTTP libraries. To learn more about this feature, [click here](LinkForTransportAdapterDocs).

Reviewer 2: ```suggestion
Out of the box, the Azure SDK for C++ supports the libcurl and WinHTTP libraries as HTTP stacks for communicating with Azure services over the network.  The SDK also provides a mechanism for `customer-implemented` *HTTP transport adapter*. To learn more about this feature, [click here](LinkForTransportAdapterDocs).
```
